### PR TITLE
Add default function example to Docs

### PR DESCRIPTION
**Pull request by JohnDeved**
## Related Issues

- #453

### CODE REVIEW

File: docs/api/decorators/prop.md [hunk 1]
```diff
@@ -64,15 +64,29 @@ Accepts Type: `any`
 
 Set a default, when no value is given at creation time.
 
-Example:
-
+Example:
```
Comment (hasezoey) [inline docs/api/decorators/prop.md:67]:
```suggestion
Example: 

```

File: docs/api/decorators/prop.md [hunk 2]
```diff
@@ -64,15 +64,29 @@ Accepts Type: `any`
...
 class Defaulted {
   @prop({ default: 'hello world' })
   public upperCase?: string; // mark as optional, because it will be defaulted
 }
 ```
 
+You may also set the `default` schema option to a function. Mongoose will execute that function and use the return value as the default. 
+
+Example: 
+```ts
```
Comment (hasezoey) [inline docs/api/decorators/prop.md:78]:
```suggestion
Example: 

```ts
```

**Review by hasezoey — CHANGES_REQUESTED** [review]
the currently style is to have an blank line before & after an code block

**Review by hasezoey — APPROVED** [review]

File: docs/api/decorators/prop.md [hunk 3]
```diff
@@ -73,6 +73,22 @@ class Defaulted {
...
 ```
 
+You may also set the `default` schema option to a function. Mongoose will execute that function and use the return value as the default. 
+
+Example: 
+
+```ts
+class Defaulted {
+  @prop({ 
+    default (this: DocumentType<Defaulted>) {
```
Comment (hasezoey) [inline docs/api/decorators/prop.md:83]:
```suggestion
    default: (this: DocumentType<Defaulted>) => {
```

from my knowledge, the one before was not valid JS/TS

Comment (JohnDeved) [reply docs/api/decorators/prop.md]: it is.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Method_definitions

Comment (JohnDeved) [reply docs/api/decorators/prop.md]: its just a shorthand for the same thing

Comment (hasezoey) [reply docs/api/decorators/prop.md]: learned a thing, thanks for providing source on this one, thought i would prefer keeping it in the same style as the other code for now

**Pull request merged**